proxy/http/HttpSM.cc

```diff
@@ -540,30 +540,35 @@ HttpSM::attach_client_session(ProxyTransaction *client_vc)
 
   // Collect log & stats information. We've already verified that the netvc is !nullptr above,
   // and netvc == ua_txn->get_netvc().
-  SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(netvc);
 
   is_internal       = netvc->get_is_internal_request();
   mptcp_state       = netvc->get_mptcp_state();
   client_tcp_reused = !(ua_txn->is_first_transaction());
 
-  if (ssl_vc != nullptr) {
+  if (auto tbs = dynamic_cast<TLSBasicSupport *>(netvc)) {
```

Comment:
Yes, the multiple `dynamic_cast` operations are a concern. One alternative would be to put a virtual method on `UnixNetVConnection` like
```
virtual TLSBasicSupport * tls_services() { return nullptr; }
```

Reply:
I'm also fine with it. One thing I don't like very much is that we'd need to have a function for each service. Is it possible to have an universal function like this?
```
T *getService<T>();
```

Reply:
Yes, but I might need to bring in a bit of meta programming support from libSWOC. You'd want something like
```
template<typename T> auto get_service_for() -> EnableForServiceTypes<T, T*> { return static_cast<T*>(this); }
```
This requires a type list backing it up, which lists all of the types for which there is a service. This compiles for those types and not for other types.

A problem is template functions/methods cannot be also `virtual`. You must choose between the cost of `dynamic_cast` or a fatter base API.

It would be possible for the mixins to provide the methods, but it would still be necessary to put a `using` in the inheriting class
```
class QUIC : public TicketService {
  using TicketService::ticker_service();
};
```

Reply:
Thinking about this more, I'm not sure why `feature_for<TLSBasicSupport>()` is much better than `tls_feature()`. It's only three methods which isn't much bloat and in the end, these feature sets are chunky and important enough that it's reasonable to promote virtual methods for them to `UnixNetVConnection`.